makefile

```diff
@@ -57,22 +57,22 @@ generate_api:
 	> ../bevy_script_api/src/generated.rs
 
 make_json_files:
-	rustup run nightly-2022-07-01 cargo rustdoc -p bevy_asset@0.8.1  --  -Zunstable-options --output-format json && \
-	rustup run nightly-2022-07-01 cargo rustdoc -p bevy_ecs@0.8.1 --  -Zunstable-options --output-format json && \
-	rustup run nightly-2022-07-01 cargo rustdoc -p bevy_pbr@0.8.1 --  -Zunstable-options --output-format json && \
-	rustup run nightly-2022-07-01 cargo rustdoc -p bevy_render@0.8.1 --  -Zunstable-options --output-format json && \
-	rustup run nightly-2022-07-01 cargo rustdoc -p bevy_math@0.8.1 --  -Zunstable-options --output-format json && \
-	rustup run nightly-2022-07-01 cargo rustdoc -p bevy_transform@0.8.1 --  -Zunstable-options --output-format json && \
-	rustup run nightly-2022-07-01 cargo rustdoc -p bevy_sprite@0.8.1 --  -Zunstable-options --output-format json && \
-	rustup run nightly-2022-07-01 cargo rustdoc -p bevy_ui@0.8.1 --  -Zunstable-options --output-format json && \
-	rustup run nightly-2022-07-01 cargo rustdoc -p bevy_animation@0.8.1 --  -Zunstable-options --output-format json && \
-	rustup run nightly-2022-07-01 cargo rustdoc -p bevy_core@0.8.1 --  -Zunstable-options --output-format json && \
-	rustup run nightly-2022-07-01 cargo rustdoc -p bevy_core_pipeline@0.8.1 --  -Zunstable-options --output-format json && \
-	rustup run nightly-2022-07-01 cargo rustdoc -p bevy_gltf@0.8.1 --  -Zunstable-options --output-format json && \
-	rustup run nightly-2022-07-01 cargo rustdoc -p bevy_hierarchy@0.8.1 --  -Zunstable-options --output-format json && \
-	rustup run nightly-2022-07-01 cargo rustdoc -p bevy_text@0.8.1 --  -Zunstable-options --output-format json && \
-	rustup run nightly-2022-07-01 cargo rustdoc -p bevy_time@0.8.1 --  -Zunstable-options --output-format json && \
-	rustup run nightly-2022-07-01 cargo rustdoc -p bevy_utils@0.8.1 --  -Zunstable-options --output-format json && \
-	rustup run nightly-2022-07-01 cargo rustdoc -p bevy_reflect@0.8.1 --  -Zunstable-options --output-format json && \
+	rustup run nightly-2022-07-01 cargo rustdoc -p bevy_asset@0.9.1  --  -Zunstable-options --output-format json && \
```

Comment:
Did running this work without any problems? If so that's fantastic, when I experimented with the migration initially It seemed like I'd have to bump the nighty version and do a lot of migrating

Reply:
No it did not, I did not do a lot of testing of this beforehand, so much testing probably needs to occur.
With my new changes, make_json_files works, but generate_api and bevy_api_gen in general does not work.

Reply:
I see now what you got with https://github.com/rust-lang/rust/issues/104651.  I have an idea for walking a tree but it will probably get messy and slow, is that fine?

Reply:
Ah I see! So for this fix, I see two possible solutions both of which are acceptable I believe:
- One would be to simply add the import paths for each item as part of the `api_gen_config.toml` file
- Another would be to as you say walk the import tree and find the first suitable public import for each item (we only use those to generate `use` statements right now so no specific import path is necessary. Probably want all imports to have a bevy prefix as well unless there is no suitable re-export by bevy)

The first solution is obviously manual work, not insane amounts but a tad annoying, the other solution is probably better in the long run.

When it comes to speed, I think as long as the whole API generation doesn't take longer than 30-60 seconds I think I am happy with that, and as long as the `messy` methods for walking the imports are clearly separated (in their own module) it should all be fine. Let's be honest `bevy_api_gen` is a whole mess in itself rn lol

Reply:
I have done this partially using the latter solution, but it doesn't seem to work with traits. It runs in about 2 seconds, which seems fine.

Reply:
Looks good to me so far